### PR TITLE
User/tracking/requests

### PR DESCRIPTION
**Pull request by HamadAlharbi1**

### CODE REVIEW

File: lib/pages/request_quotations_page.dart [hunk 1]
```diff
@@ -12,17 +15,59 @@ class RequestQuotations extends StatefulWidget {
...
+
+  Future<void> errorShow(var e) async {
+    final snackBar = SnackBar(
+      content: Text('error:$e'.i18n()),
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  requestQuotation() async {
+    final requestId = const Uuid().v4();
```
Comment (mutairibassam) [inline lib/pages/request_quotations_page.dart:50]: request id shouldn't be randomly generated due to id conflict. request id should be incremented by one starting from 1000.

File: lib/pages/request_quotations_page.dart [hunk 2]
```diff
@@ -12,17 +15,59 @@ class RequestQuotations extends StatefulWidget {
...
+      await FirebaseFirestore.instance.collection('RequestQuotations').doc(requestId).set({
+        'nationalId': nationalIdNumberController.text,
+        'birthDate': birthDateController.text,
+        'carSerialNumber': carSerialNumberController.text,
+        'userId': uid,
+        'status': "under_review",
+        'phoneNumber': phoneNumber,
+        'insuranceAmount': 1.5,
+        'requestId': requestId,
+        'requestType': "Request_Quotations",
```
Comment (mutairibassam) [inline lib/pages/request_quotations_page.dart:62]: Request type should be written as enums to unify types and avoid typos. Moreover, it should be simplified either using numbers [1,2,3,4] or [quotation, ...] without request.

File: lib/pages/tracking_requests.dart [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  @override
+  State<TrackingRequests> createState() => _TrackingRequestsState();
+}
+
+class _TrackingRequestsState extends State<TrackingRequests> {
+  List<Quotations> quotations = [];
+  listenRequestQuotations() async {
+    final collection =
+        FirebaseFirestore.instance.collection('RequestQuotations').where('userId', isEqualTo: '$uid').snapshots();
```
Comment (mutairibassam) [inline lib/pages/tracking_requests.dart:20]:
```suggestion
        FirebaseFirestore.instance.collection('quotations').where('userId', isEqualTo: '$uid').snapshots();
```

File: pubspec.yaml [hunk 4]
```diff
@@ -53,6 +53,7 @@ dependencies:
 
     flutter_localizations:
         sdk: flutter
+    uuid: ^4.1.0
```
Comment (mutairibassam) [inline pubspec.yaml:56]: missing docs.

**Review by mutairibassam — CHANGES_REQUESTED** [review]

File: lib/app.dart [hunk 5]
```diff
@@ -58,3 +60,12 @@ MaterialColor createMaterialColor(Color color) {
...
 }
+
+Widget buildFirstWidget() {
+  final User? firebaseUser = FirebaseAuth.instance.currentUser;
+  if (firebaseUser == null) {
+    return const RegistrationPage();
+  } else {
+    return const HomePage();
+  }
+}
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/app.dart]: i try this but its not working ..

File: lib/app.dart [hunk 6]
```diff
@@ -33,7 +35,7 @@ class MyApp extends StatelessWidget {
       },
       theme: themeData,
       debugShowCheckedModeBanner: false,
-      home: const RegistrationPage(),
+      home: buildFirstWidget(),
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/app.dart]: name changed  to  checkUserAuthentication

File: pubspec.yaml (hunk 4 above)
Comment (HamadAlharbi1) [reply pubspec.yaml:57]: done

File: lib/pages/request_quotations_page.dart (hunk 1 above)
Comment (HamadAlharbi1) [reply lib/pages/request_quotations_page.dart]: done

File: lib/pages/request_quotations_page.dart [hunk 7]
```diff
@@ -12,17 +15,59 @@ class RequestQuotations extends StatefulWidget {
...
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  Future<void> errorShow(var e) async {
+    final snackBar = SnackBar(
+      content: Text('error:$e'.i18n()),
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/pages/request_quotations_page.dart]: done

File: lib/pages/registration_page.dart [hunk 8]
```diff
@@ -64,16 +96,16 @@ class _RegistrationPageState extends State<RegistrationPage> {
...
-                  width: 200,
-                  color: const Color.fromARGB(255, 180, 180, 180),
-                  child: const Center(
-                      child: Text(
-                    "Logo",
-                    style: TextStyle(fontSize: 30),
-                  )),
+                  child: Image.network(
+                      'https://cdn.discordapp.com/attachments/1083124198827888830/1159214780209434726/image.png?ex=653035e4&is=651dc0e4&hm=e703873e965917a1d19716ea732c2f6383908ed6d89c33444c09213aa64d5dcf&'),
+                ),
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/pages/registration_page.dart]: done

File: lib/pages/registration_page.dart [hunk 9]
```diff
@@ -17,38 +17,70 @@ class RegistrationPage extends StatefulWidget {
...
-                // print('Invalid OTP. Please try again.');
-              }
-            },
-          ),
+  displayError(Object e) {
+    final snackBar = SnackBar(
+      content: Text('Error: $e'),
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/pages/registration_page.dart]: done

File: lib/pages/home_page.dart [hunk 10]
```diff
@@ -32,6 +33,11 @@ class _HomePageState extends State<HomePage> {
                   Navigator.push(context, MaterialPageRoute(builder: (context) => const RequestRefund()));
                 },
                 text: "Request_Refund".i18n()),
+            HomePageActionsContainer(
+                onPressed: () {
+                  Navigator.push(context, MaterialPageRoute(builder: (context) => const TrackingRequests()));
+                },
+                text: "Tracking_Requests".i18n()),
```
[the remark this answers is not in the document]
Comment (HamadAlharbi1) [reply lib/pages/home_page.dart]: done

File: lib/pages/tracking_requests.dart (hunk 3 above)
Comment (HamadAlharbi1) [reply lib/pages/tracking_requests.dart]: done

**Review by mutairibassam — CHANGES_REQUESTED** [review]

**Pull request merged**